### ISSUE TITLE
Try out factor graph with GD & GP

### ISSUE THREAD

**Issue by jannis-baum** [jannis-baum/hpi-kaki#1]

**Comment by jannis-baum** [comment]
# Next steps

(if it comes to it)

- test🙃🫠 before trying to model function relationship
- gradient descent: is it okay to go by change in cost vs. absdiff of parameters? i guess so
- run GP on outputs; approaches to variance problems:
  - consider fixing variance to 1 value for each series in model, training variance with GPs and multiplying (?) for final inference
  - heteroskedastic model
  - sample normal distributions and use samples to train normal GP (way more training data, probably bad & too slow)
  - don't use a GP at all; might be fair to assume some function shape (e.g. value is lowest at 0, variance increases with abs of mean) that could be described by ~3 parameters
  - try out using 2 GPs and factoring uncertainty of both into predicted variance
